modules/linear_boltzmann_solvers/executors/pi_keigen_smm.cc

```diff
@@ -0,0 +1,1122 @@
+}
+
+PowerIterationKEigenSMM::PowerIterationKEigenSMM(const InputParameters& params)
+  : PowerIterationKEigen(params),
+    dimension_(0),
+    psi_new_local_(lbs_solver_.PsiNewLocal()),
+    accel_pi_max_its_(params.GetParamValue<unsigned int>("accel_pi_max_its")),
+    accel_pi_k_tol_(params.GetParamValue<double>("accel_pi_k_tol")),
+    accel_pi_verbose_(params.GetParamValue<bool>("accel_pi_verbose")),
+    diffusion_sdm_name_(params.GetParamValue<std::string>("diff_sdm")),
+    diffusion_l_max_its_(params.GetParamValue<int>("diff_l_max_its")),
+    diffusion_l_abs_tol_(params.GetParamValue<double>("diff_l_abs_tol")),
+    diffusion_petsc_options_(params.GetParamValue<std::string>("diff_petsc_options")),
+    diffusion_verbose_(params.GetParamValue<bool>("diff_verbose"))
+{
+  ghosts_required_ = diffusion_sdm_name_ == "pwlc";
+  lbs_solver_.Options().save_angular_flux = true;
+  if (lbs_solver_.Groupsets().size() != 1)
+    throw std::logic_error("The SMM k-eigenvalue executor is only implemented for "
+                           "problems with a single groupset.");
+}
+
+void
+PowerIterationKEigenSMM::Initialize()
+{
+  PowerIterationKEigen::Initialize();
+
+  // Shorthand information
+  const auto& grid = lbs_solver_.Grid();
+  const auto& pwld = lbs_solver_.SpatialDiscretization();
+  const auto& phi_uk_man = lbs_solver_.UnknownManager();
+  const auto num_groups = lbs_solver_.NumGroups();
+  const auto num_gs_groups = front_gs_.groups_.size();
+
+  // Specialized unit cell matrices
+  dimension_ = lbs_solver_.Grid().Dimension();
+  ComputeAuxiliaryUnitCellMatrices();
+
+  // Create PWLC structure, if needed
+
+  if (diffusion_sdm_name_ == "pwlc")
+  {
+    pwlc_ptr_ = PieceWiseLinearContinuous::New(grid);
+    ghost_info_ = MakePWLDGhostInfo(pwld, phi_uk_man);
+  }
+
+  //======================================== Create tensor structure
+
+  for (int g = 0; g < num_groups; ++g)
+    tensor_uk_man_.AddUnknown(UnknownType::VECTOR_N, dimension_ * dimension_);
+
+  const auto local_size = pwld.GetNumLocalDOFs(tensor_uk_man_);
+  const auto global_size = pwld.GetNumGlobalDOFs(tensor_uk_man_);
+  const auto ghost_ids = MakePWLDGhostIndices(pwld, tensor_uk_man_);
+  tensors_ = std::make_unique<GhostedParallelSTLVector>(
+    local_size, global_size, ghost_ids, opensn::mpi_comm);
+
+  //======================================== Create diffusion solver
+
+  // Create the unknown manager
+  UnknownManager diff_uk_man;
+  diff_uk_man.AddUnknown(UnknownType::VECTOR_N, num_gs_groups);
+
+  // Create the diffusion boundaries
+  for (const auto& [bid, bc] : lbs_solver_.BoundaryPreferences())
+    if ((bc.type == BoundaryType::ISOTROPIC) or (bc.type == BoundaryType::ARBITRARY))
+      throw std::logic_error("Only vacuum and reflective boundaries are valid for "
+                             "k-eigenvalue problems.");
+
+  const auto bcs = TranslateBCs(lbs_solver_.SweepBoundaries(),
+                                /*vacuum_bcs_are_dirichlet=*/false);
```

Comment:
Remove `/*vacuum_bcs_are_dirichlet=*/`.

Reply:
I actually thought it was a clever way of commenting something related to the specifics of the method. I would very much prefer to keep such comments.

Reply:
I have no preference. My last commit removes these. I can revert later if necessary.

Reply:
I prefer to get these.

Reply:
@andrsd How firm are you on removing these?

Reply:
They are one-off thing. How do you determine when to use them and when not? You certainly don't call other methods using this style. It reminds me of keyword params from Python, i.e. this does not belong into C++, IMO. And there should be doco comments at the method declarations to explain these things, so user code should not need to do this at all.
In my book, these are good reasons against doing this.

Reply:
I think those are fair points. I will leave them out and not revert the commit removing them.